syntax = "proto3";

option java_outer_classname = "FragmentResolverProto";

message FragmentResolverModelRequest {
  // float encoded in little endian
  bytes audioSamplesChannel1 = 1;
}

message ResolvedTransformer {
  enum Type {
    SILENCE = 0;
  }

  Type type = 1;
  int64 silenceDurationUs = 2;
}

message ResolvedFragment {
  int64 startUs = 1;
  int64 endUs = 2;
  ResolvedTransformer transformer = 3;
}

message FragmentResolverModelResponse {
  repeated ResolvedFragment fragments = 1;
}

message FragmentResolverModelConfig {
  int32 sampleRate = 1;
}